core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/InternalBinaryRange.java

```diff
@@ -166,6 +167,24 @@ public String getToAsString() {
             return to == null ? null : format.format(to);
         }
 
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            Bucket bucket = (Bucket) o;
+
+            if (docCount != bucket.docCount) return false;
+            return Objects.equals(key, bucket.key) &&
+                Objects.equals(from, bucket.from) &&
+                Objects.equals(to, bucket.to) &&
+                Objects.equals(aggregations, bucket.aggregations);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(keyed, format, docCount, key, from, to, aggregations);
```

Comment:
keyed and format are in the hashcode but not in equals. I think it is fine not to consider them in equals since they only have an impact on formatting, but then hashCode should not use them either?